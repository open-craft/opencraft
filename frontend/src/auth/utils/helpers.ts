import { V2Api } from 'global/api';

export const checkAuthAndRefreshToken = async () => {
  const refreshToken = localStorage.getItem('token_refresh');
  let response = null;

  if (refreshToken) {
    try {
      response = await V2Api.authRefreshCreate({
        data: { refresh: refreshToken }
      });
    } catch (e) {
      return false;
    }
  }

  if (response) {
    localStorage.setItem('token_access', response.access);
    return true;
  }

  return false;
};
