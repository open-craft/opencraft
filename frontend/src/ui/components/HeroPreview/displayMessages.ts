const messages = {
  edit: {
    defaultMessage: 'Edit',
    description: ''
  }
};

export default messages;
