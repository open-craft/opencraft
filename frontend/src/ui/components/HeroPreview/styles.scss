.hero-preview {
  background-color: #9c9c9c;
  display:block;

  .outer-wrapper {
    padding: 60px 20px;
    /*
    FIXME: This doesn't give the actual representation of how the background image will look like in the LMS.
    */
    background-size: cover;
    background-position: center;
  }

  .title {
    background: #fff;
    padding: 40px 30px;
  }

  h1 {
    text-align: left;
    color: black;
    font: normal 1.6em/0.7em "Open Sans","Helvetica Neue", Helvetica,Arial,sans-serif;
  }

  p {
    font: italic 1em/1.2em Georgia,Cambria,"Times New Roman",Times,serif;
    color: #646464;
  }
}
