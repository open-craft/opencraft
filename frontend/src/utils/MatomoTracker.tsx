import React, { ReactNode } from 'react';
import {
  createInstance,
  MatomoProvider,
  useMatomo
} from '@datapunt/matomo-tracker-react';
import {
  MATOMO_ALIAS_DOMAIN,
  MATOMO_BASE_URL,
  MATOMO_MY_DOMAIN,
  MATOMO_SITE_ID
} from '../global/constants';

interface MatomoTrackerProps {
  children: ReactNode;
}

interface MatomoAdditionalConfiguration {
  [propName: string]: any;
}

interface MatomoConfiguration {
  urlBase: string;
  siteId: number;
  configurations?: MatomoAdditionalConfiguration;
}

interface MatomoUserIdTrackerProps {
  userId: string;
}

/*
 * This wrapper component initializes the Matomo tracking by wrapping the children
 * elements, if the tracking is enabled by providing a valid value for the 'MATOMO_BASE_URL'
 * constant. It allows tracking page views, events and the user id, when available.
 *
 * The 'MATOMO_ALIAS_DOMAIN' constant allows the cross-domain linking of another site with
 * this site so that they are treated as the same site in Matomo.
 */
export const MatomoTracker: React.FC<MatomoTrackerProps> = (
  props: MatomoTrackerProps
) => {
  let element = props.children;
  if (MATOMO_BASE_URL) {
    const matomoConfiguration: MatomoConfiguration = {
      urlBase: MATOMO_BASE_URL,
      siteId: parseInt(MATOMO_SITE_ID, 10)
    };
    if (MATOMO_ALIAS_DOMAIN) {
      const myDomain = `*.${MATOMO_MY_DOMAIN}`;
      const aliasDomain = `*.${MATOMO_ALIAS_DOMAIN}`;
      matomoConfiguration.configurations = {
        enableCrossDomainLinking: true,
        setDomains: [myDomain, aliasDomain]
      };
    }
    element = (
      <MatomoProvider value={createInstance(matomoConfiguration)}>
        {props.children}
      </MatomoProvider>
    );
  }
  return <>{element}</>;
};

/**
 * This is a component passes the provided user id to Matomo if Matomo tracking is enabled.
 * The 'pushInstruction' function exposed by the 'useMatomo()' hook is used to do this.
 *
 * This component is needed as all the components in the registration process are classes and hence
 * the 'yseMatomo' hook cannot be directly used in those components.
 */
export const MatomoUserIdTracker: React.FC<MatomoUserIdTrackerProps> = (
  props: MatomoUserIdTrackerProps
) => {
  const { pushInstruction } = useMatomo();
  if (MATOMO_BASE_URL && props.userId) {
    pushInstruction('setUserId', props.userId);
  }
  return null;
};
