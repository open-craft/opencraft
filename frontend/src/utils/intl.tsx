import React from "react";
import { FormattedMessage, MessageDescriptor } from "react-intl";

type MessageMap = { [k: string]: Omit<MessageDescriptor, 'id'> };

interface Props<T, K extends keyof T> {
    id: K;
    messages: T;
    values?: any;
}


export const WrappedMessage = (props: Props<MessageMap, string>) => {
    const {messages, id, ...values} = props;
    return (
        <FormattedMessage  {...messages[id]} id={id} {...values}/>
    );
};
