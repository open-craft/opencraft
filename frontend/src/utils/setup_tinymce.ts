import 'tinymce/tinymce';
import 'tinymce/themes/silver';
import 'tinymce/skins/ui/oxide/skin.min.css';
import 'tinymce/skins/ui/oxide/content.min.css';
import 'tinymce/plugins/advlist';
import 'tinymce/plugins/autolink';
import 'tinymce/plugins/lists';
import 'tinymce/plugins/link';
import 'tinymce/plugins/image';
import 'tinymce/plugins/charmap';
import 'tinymce/plugins/print';
import 'tinymce/plugins/preview';
import 'tinymce/plugins/anchor';
import 'tinymce/plugins/searchreplace';
import 'tinymce/plugins/visualblocks';
import 'tinymce/plugins/code';
import 'tinymce/plugins/fullscreen';
import 'tinymce/plugins/insertdatetime';
import 'tinymce/plugins/media';
import 'tinymce/plugins/table';
import 'tinymce/plugins/paste';
import 'tinymce/plugins/help';
import 'tinymce/plugins/wordcount';
