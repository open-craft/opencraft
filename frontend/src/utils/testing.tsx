import { createMemoryHistory } from 'history';
import * as React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import * as renderer from 'react-test-renderer';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { createRootReducer } from '../global/reducers';

export const setupComponentForTesting = (
  reactContent: JSX.Element,
  storeContents = {}
) => {
  const middleware = applyMiddleware(thunk);

  const store = createStore(
    createRootReducer(createMemoryHistory()),
    storeContents,
    middleware
  );

  return renderer.create(
    <IntlProvider textComponent={React.Fragment} locale="en">
      <Provider store={store}>
        <MemoryRouter>{reactContent}</MemoryRouter>
      </Provider>
    </IntlProvider>
  );
};
