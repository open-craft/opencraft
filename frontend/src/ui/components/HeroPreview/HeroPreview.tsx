import * as React from 'react';
import './styles.scss';

interface HeroPreviewProps {
  homePageHeroTitleColor: undefined | string;
  homePageHeroSubtitleColor: undefined | string;
  homepageOverlayHtml: undefined | string;
  heroCoverImage: undefined | string;
}

export const HeroPreview: React.FC<HeroPreviewProps> = (
  props: HeroPreviewProps
) => {
  let titleStyles = {};
  let subtitleStyles = {};
  let containerStyles = {};
  let matched;
  const heroTextRegex = /^<h1>(.*)<\/h1><p>(.*)<\/p>$/;

  if (props.homePageHeroTitleColor) {
    titleStyles = {
      color: props.homePageHeroTitleColor
    };
  }
  if (props.homePageHeroSubtitleColor) {
    subtitleStyles = {
      color: props.homePageHeroSubtitleColor
    };
  }
  if (props.homepageOverlayHtml) {
    matched = heroTextRegex.exec(props.homepageOverlayHtml as string);
  }
  if (props.heroCoverImage) {
    containerStyles = {
      backgroundImage: `url("${props.heroCoverImage}")`
    };
  }
  return (
    <div className="hero-preview">
      <div className="outer-wrapper" style={containerStyles}>
        <div className="title">
          <div className="heading-group">
            <h1 style={titleStyles}>{matched ? matched[1] : ''}</h1>
            <p style={subtitleStyles}>{matched ? matched[2] : ''}</p>
          </div>
        </div>
      </div>
    </div>
  );
};
