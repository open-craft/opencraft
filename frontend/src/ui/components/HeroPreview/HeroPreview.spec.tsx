import React from 'react';
import { setupComponentForTesting } from "utils/testing";
import { HeroPreview } from './HeroPreview';

describe("Hero preview component", function() {
    it('renders without crashing', () => {
        const tree = setupComponentForTesting(<HeroPreview />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('renders without crashing when instance has no theme set up', () => {
          const tree = setupComponentForTesting(
          <HeroPreview
            homepageOverlayHtml="<h1>Welcome to Test instance</h1><p>It works! Powered by Open edX®</p>"
            heroCoverImage="https://example.com/image.png"
          />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('renders without crashing when instance has no home page overlay html provided', () => {
      const tree = setupComponentForTesting(
          <HeroPreview
            heroCoverImage="https://example.com/image.png"
            homePageHeroTitleColor='#000'
            homePageHeroSubtitleColor='#999'
          />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('renders without crashing when instance has no cover image provided', () => {
      const tree = setupComponentForTesting(
          <HeroPreview
            homePageHeroTitleColor='#000'
            homePageHeroSubtitleColor='#999'
            homepageOverlayHtml="<h1>Welcome to Test instance</h1><p>It works! Powered by Open edX®</p>"
          />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('renders without crashing when the instance only has the cover image provided', () => {
      const tree = setupComponentForTesting(
          <HeroPreview
            heroCoverImage="https://example.com/image.png"
          />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('renders without crashing when the instance only has the home page overlay html provided', () => {
      const tree = setupComponentForTesting(
          <HeroPreview
            homepageOverlayHtml="<h1>Welcome to Test instance</h1><p>It works! Powered by Open edX®</p>"
          />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('renders without crashing when the instance only has the theme config set up', () => {
      const tree = setupComponentForTesting(
          <HeroPreview
            homePageHeroTitleColor='#000'
            homePageHeroSubtitleColor='#999'
          />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('renders without crashing when the instance only has the hero title color set up', () => {
      const tree = setupComponentForTesting(
          <HeroPreview
            homePageHeroTitleColor='#000'
          />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('renders without crashing when the instance only has the hero subtitle color set up', () => {
      const tree = setupComponentForTesting(
          <HeroPreview
            homePageHeroSubtitleColor='#000'
          />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
