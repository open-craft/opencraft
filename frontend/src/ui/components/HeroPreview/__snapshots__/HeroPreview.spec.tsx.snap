// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero preview component renders without crashing 1`] = `
<div
  className="hero-preview"
>
  <div
    className="outer-wrapper"
    style={Object {}}
  >
    <div
      className="title"
    >
      <div
        className="heading-group"
      >
        <h1
          style={Object {}}
        >
          
        </h1>
        <p
          style={Object {}}
        >
          
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Hero preview component renders without crashing when instance has no cover image provided 1`] = `
<div
  className="hero-preview"
>
  <div
    className="outer-wrapper"
    style={Object {}}
  >
    <div
      className="title"
    >
      <div
        className="heading-group"
      >
        <h1
          style={
            Object {
              "color": "#000",
            }
          }
        >
          Welcome to Test instance
        </h1>
        <p
          style={
            Object {
              "color": "#999",
            }
          }
        >
          It works! Powered by Open edX®
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Hero preview component renders without crashing when instance has no home page overlay html provided 1`] = `
<div
  className="hero-preview"
>
  <div
    className="outer-wrapper"
    style={
      Object {
        "backgroundImage": "url(\\"https://example.com/image.png\\")",
      }
    }
  >
    <div
      className="title"
    >
      <div
        className="heading-group"
      >
        <h1
          style={
            Object {
              "color": "#000",
            }
          }
        >
          
        </h1>
        <p
          style={
            Object {
              "color": "#999",
            }
          }
        >
          
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Hero preview component renders without crashing when instance has no theme set up 1`] = `
<div
  className="hero-preview"
>
  <div
    className="outer-wrapper"
    style={
      Object {
        "backgroundImage": "url(\\"https://example.com/image.png\\")",
      }
    }
  >
    <div
      className="title"
    >
      <div
        className="heading-group"
      >
        <h1
          style={Object {}}
        >
          Welcome to Test instance
        </h1>
        <p
          style={Object {}}
        >
          It works! Powered by Open edX®
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Hero preview component renders without crashing when the instance only has the cover image provided 1`] = `
<div
  className="hero-preview"
>
  <div
    className="outer-wrapper"
    style={
      Object {
        "backgroundImage": "url(\\"https://example.com/image.png\\")",
      }
    }
  >
    <div
      className="title"
    >
      <div
        className="heading-group"
      >
        <h1
          style={Object {}}
        >
          
        </h1>
        <p
          style={Object {}}
        >
          
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Hero preview component renders without crashing when the instance only has the hero subtitle color set up 1`] = `
<div
  className="hero-preview"
>
  <div
    className="outer-wrapper"
    style={Object {}}
  >
    <div
      className="title"
    >
      <div
        className="heading-group"
      >
        <h1
          style={Object {}}
        >
          
        </h1>
        <p
          style={
            Object {
              "color": "#000",
            }
          }
        >
          
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Hero preview component renders without crashing when the instance only has the hero title color set up 1`] = `
<div
  className="hero-preview"
>
  <div
    className="outer-wrapper"
    style={Object {}}
  >
    <div
      className="title"
    >
      <div
        className="heading-group"
      >
        <h1
          style={
            Object {
              "color": "#000",
            }
          }
        >
          
        </h1>
        <p
          style={Object {}}
        >
          
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Hero preview component renders without crashing when the instance only has the home page overlay html provided 1`] = `
<div
  className="hero-preview"
>
  <div
    className="outer-wrapper"
    style={Object {}}
  >
    <div
      className="title"
    >
      <div
        className="heading-group"
      >
        <h1
          style={Object {}}
        >
          Welcome to Test instance
        </h1>
        <p
          style={Object {}}
        >
          It works! Powered by Open edX®
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Hero preview component renders without crashing when the instance only has the theme config set up 1`] = `
<div
  className="hero-preview"
>
  <div
    className="outer-wrapper"
    style={Object {}}
  >
    <div
      className="title"
    >
      <div
        className="heading-group"
      >
        <h1
          style={
            Object {
              "color": "#000",
            }
          }
        >
          
        </h1>
        <p
          style={
            Object {
              "color": "#999",
            }
          }
        >
          
        </p>
      </div>
    </div>
  </div>
</div>
`;
